#Name: Syed Sameer Rizvi

#AICP Task Week 5

In [4]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio
from wordcloud import WordCloud
pio.templates.default = "plotly_white"

Q.1: Show column names and have a look at their info.


In [8]:
df = pd.read_csv("instagram_data.csv", encoding='latin-1')

print(df.columns)

print("Info:")
print(df.info)

Index(['Impressions', 'From Home', 'From Hashtags', 'From Explore',
       'From Other', 'Saves', 'Comments', 'Shares', 'Likes', 'Profile Visits',
       'Follows', 'Caption', 'Hashtags'],
      dtype='object')
Info:
<bound method DataFrame.info of      Impressions  From Home  From Hashtags  From Explore  From Other  Saves  \
0           3920       2586           1028           619          56     98   
1           5394       2727           1838          1174          78    194   
2           4021       2085           1188             0         533     41   
3           4528       2700            621           932          73    172   
4           2518       1704            255           279          37     96   
..           ...        ...            ...           ...         ...    ...   
114        13700       5185           3041          5352          77    573   
115         5731       1923           1368          2266          65    135   
116         4139       1133           15

Q.2: Show the descriptive statistics of the data

In [9]:
df.describe()

Impressions     From Home  From Hashtags  From Explore   From Other  \
count    119.000000    119.000000     119.000000    119.000000   119.000000   
mean    5703.991597   2475.789916    1887.512605   1078.100840   171.092437   
std     4843.780105   1489.386348    1884.361443   2613.026132   289.431031   
min     1941.000000   1133.000000     116.000000      0.000000     9.000000   
25%     3467.000000   1945.000000     726.000000    157.500000    38.000000   
50%     4289.000000   2207.000000    1278.000000    326.000000    74.000000   
75%     6138.000000   2602.500000    2363.500000    689.500000   196.000000   
max    36919.000000  13473.000000   11817.000000  17414.000000  2547.000000   

             Saves    Comments      Shares       Likes  Profile Visits  \
count   119.000000  119.000000  119.000000  119.000000      119.000000   
mean    153.310924    6.663866    9.361345  173.781513       50.621849   
std     156.317731    3.544576   10.089205   82.378947       87.088402   
min      22.000000    0.000000    0.000000   72.000000        4.000000   
25%      65.000000    4.000000    3.000000  121.500000       15.000000   
50%     109.000000    6.000000    6.000000  151.000000       23.000000   
75%     169.000000    8.000000   13.500000  204.000000       42.000000   
max    1095.000000   19.000000   75.000000  549.000000      611.000000   

          Follows  
count  119.000000  
mean    20.756303  
std     40.921580  
min      0.000000  
25%      4.000000  
50%      8.000000  
75%     18.000000  
max    260.000000

Q.3: Check if your data contains any missing values

In [10]:
df.isnull()

Impressions  From Home  From Hashtags  From Explore  From Other  Saves  \
0          False      False          False         False       False  False   
1          False      False          False         False       False  False   
2          False      False          False         False       False  False   
3          False      False          False         False       False  False   
4          False      False          False         False       False  False   
..           ...        ...            ...           ...         ...    ...   
114        False      False          False         False       False  False   
115        False      False          False         False       False  False   
116        False      False          False         False       False  False   
117        False      False          False         False       False  False   
118        False      False          False         False       False  False   

     Comments  Shares  Likes  Profile Visits  Follows  Caption  Hashtags  
0       False   False  False           False    False    False     False  
1       False   False  False           False    False    False     False  
2       False   False  False           False    False    False     False  
3       False   False  False           False    False    False     False  
4       False   False  False           False    False    False     False  
..        ...     ...    ...             ...      ...      ...       ...  
114     False   False  False           False    False    False     False  
115     False   False  False           False    False    False     False  
116     False   False  False           False    False    False     False  
117     False   False  False           False    False    False     False  
118     False   False  False           False    False    False     False  

[119 rows x 13 columns]

No charts were generated by quickchart


Q.4:  When you start exploring your data, always start by exploring the main feature of your data. For
example, as we are working on a dataset based on Instagram Reach, we should start by exploring the
feature that contains data about reach. In our data, the Impressions column contains the data about the
reach of an Instagram post. So let’s have a look at the distribution of the Impressions:

#Summary: Distribution of Impressions
The Image that was provided shows that the most frequent imprrsion count lies in between 5k to 10k. Also there is a good amount of impressions that fall in between 15k t0 20k. The data diminishes after 20k.

Q.5: Have a look at the number of impressions on each post over time as shown below


#Summary: Impressions over time

The image shows a line graph, so it is diffcult to say much about the trends because the x-axis does not have ticks labeled. However, according to my observations, there is a period of incerasing impressions as well as period of decreasing impressions.

Q.6: Have a look at all the metrics like Likes, Saves, and Follows from each post over time as
shown below

#Summary: Metrics over time

The image provided shows about engagements (likes,saves and follows). This line graph shows that the user's overall saves, likes and follows take place during those dates we can see the the number of saves of a post/reel spikes in between 40 to 60 and also afterr 100 on the date. However, this shows that the user's save more than by liking or following the creator of that post/reel.

Q.7: Have a look at the distribution of reach from different sources as shown below


#Summary: Reach from different sources
The image composes of a pie chart which shows the distribution of how many times an instagram post/reel was seen from various fields like from houses, from hastags, from explore and from other. The largest slice of the pie chart is 44.1% of "From home". This slice suggests that nearly half of the impressions came from people at home.  


Q.8: Have a look at the distribution of engagement sources as shown below

#Summary: Engagement Sources

This here shows how people engaged with the posts either by liking or saving or sharing or commenting the post. In the image, the largest slice is 50.6 of "Likes". This shows that people engage with the posts by giving a like.

Q.9: Have a look at the relationship between the number of profile visits and follows as shown
below

#Summary: Profile Visits vs Follows

This line graph shows that there are more profile visits than follows throughout the time. There is an upward trend in the profile visits over time than follows over time. Without the eexact units on the x-axis I cannot say for sure that by visiting the profile would genrally result in a follow or how many profiles that are visited would eventually end in one of them being followed.

Q.10: Have a look at the type of hashtags used in the posts using a wordcloud as shown below

#Summary: Hashtag Word Cloud

The image composed of a word cloud visualization and this shows that the most frequent words are related to data science and programming including "python", "dataanalysis", "datascience", and "machinelearning". This also shows that a source that discusses data science or programming.

Q.11: Have a look at the correlation between all the features as shown below

#Summary: Correlation Matrix

Here are some observations about the correlations in the matrix:

Follows has a strong positive correlation with Likes, Shares, and Comments. This means that when there are more Follows, there are also more likes, shares, and comments.

Impressions has a negative correlation with all of the other metrics follows, likes, saves, comments, shares. This means that when there are more Impressions, there are fewer of the other metrics.

Saves has a weak positive correlation with Shares and Comments. This means that there might be a slight increase in Shares and Comments when there are more Saves.

From Explore and From Hashtags both have a weak positive correlation with all of the engagement metrics follows, likes, saves, comments, shares This means that posts that are seen from Explore and Hashtags tend to have a little more engagement.

Q.12: Have a look at the distribution of hashtags to see which hashtag is used the most in all the
posts as shown below

#Summary: Distribution of Hashtags

The bar graph composes of most used hashtags on an instagram post and the most commonly used hashtag is related to programming like "#thecleverprogramming"  and "#pythonprogramming" (120)

Q.13: Have a look at the distribution of likes and impressions received from the presence of each
hashtag on the post as shown below


#Summary: Likes distribution for each hashtag

The bar graph shows that the number of likes a post recieves with a particular hashtag in this case a username like "Amakhawal" hashtag has about 20k likes and the 15k likes consisted of hashtags related to "Data Science" and "Programming".

#Summary: Impressions distribution for each hashtag

The bar graph shows that the number of impressions on a post recieves with a particular hashtag in this case a username like "Amakhawal" hashtag has about more than 600k impressions and in between 400k to 600k impressions on a post consisted of hashtags related to "Data Science" and "Programming".